### PR TITLE
Fix #5809: Support Steam RCT1 file layout

### LINKED ISSUE #5809: Support Steam RCT1 file layout
Steam has its RCT1 files in a sorta weird structure: it effectively does a not-full install, and then keeps the files that would be on the disk in a subdirectory. [A full directory listing of a fresh install is attached.](https://github.com/OpenRCT2/OpenRCT2/files/1127508/dir.txt) As a result, OpenRCT2 doesn't recognize the directory as a valid RCT1 folder. It seems to load the scenarios, but not the music and possibly not some other things, I'm not entirely sure

Though it does of course work fine if you copy files from the 'disc' data folder to the main data folder, it'd be a nice enhancement to not have to do so.

### PR DESCRIPTION
**Pull request by IntelOrca**
Checks RCTdeluxe_install sub directory for csg1.1 and csg1i.dat when loading CSG images.

### CODE REVIEW

File: src/openrct2/drawing/Sprite.cpp [hunk 1]
```diff
@@ -177,6 +177,34 @@ void mask_scalar(sint32 width, sint32 height, const uint8 * RESTRICT maskSrc, co
...
+    // In the CD version, it's called csg1.1 on the CD and csg1.dat on the disk.
+    // In the GOG version, it's always called csg1.1.
+    // In the Steam version, it's always called csg1.dat.
+    auto path = Path::ResolveCasing(Path::Combine(gConfigGeneral.rct1_path, "Data", "csg1.1"));
+    if (path.empty())
+    {
+        path = Path::ResolveCasing(Path::Combine(gConfigGeneral.rct1_path, "Data", "csg1.dat"));
+        if (path.empty())
+        {
+            path = Path::ResolveCasing(Path::Combine(gConfigGeneral.rct1_path, "RCTdeluxe_install", "Data", "csg1.1"));
```
Comment (Gymnasiast) [inline src/openrct2/drawing/Sprite.cpp]: In the comment, it says this file is always called csg1.dat in Steam. Not csg1.1.

Comment (Gymnasiast) [inline src/openrct2/drawing/Sprite.cpp]: I don't think this, or the comment, or neither are correct. But I want to investigate some stuff.

Comment (IntelOrca) [reply src/openrct2/drawing/Sprite.cpp]: The comment is probably not accurate.
The path check is based on the [steam directory listing](https://steamdb.info/depot/285311/).

Can you take over the PR?

Comment (Gymnasiast) [reply src/openrct2/drawing/Sprite.cpp]: Looking at the listing, it seems like Steam works like the disk version: the CSG1.DAT file is present in the "hard drive" folder, the CSG1.1 file is present on the "CD" folder. This means that the third check is unnecessary, as we expect the user to use the "hard drive" folder anyway.

**Comment by Gymnasiast** [comment]
Ok, I have made the changes I thought were necessary. It still works with my installation as before, but I'd like someone with a Steam installation to verify it now works with the unmodified Steam version of RCT1 as well. Set your path to the folder _containing_ `RCTdeluxe_install`, not the `RCTdeluxe_install` folder itself!

**Comment by deurklink** [comment]
I can confirm that it works with RCT1 steam installation. I removed csg1.1 and csg1i.dat from (which i previously copied) from the Data folder and i can still see the RCT1 style map edges. This means it must read them from the RCTdeluxe_install/Data folder.

**Pull request merged**